mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -3543,6 +3551,8 @@ convertTopLevelTargetOp(Operation *op, llvm::IRBuilderBase &builder,
                               LLVM::ModuleTranslation &moduleTranslation) {
   if (isa<omp::TargetOp>(op))
     return convertOmpTarget(*op, builder, moduleTranslation);
+  if (isa<omp::TargetDataOp>(op))
+    return convertOmpTargetData(op, builder, moduleTranslation);
   bool interrupted =
       op->walk<WalkOrder::PreOrder>([&](omp::TargetOp targetOp) {
           if (failed(convertOmpTarget(*targetOp, builder, moduleTranslation)))
```

Comment:
Wouldn't we need to find nested `omp.target_data` ops here as well?

Reply:
Yes, I forgot this wasn't a recursive function, so there should be a case for the TargetDataOp as well.